python/tank/authentication/login_dialog.py

```diff
@@ -583,12 +571,18 @@ def _toggle_web(self, method_selected=None):
             not self._query_task.unified_login_flow2_enabled
         )
         self.ui.button_options.setVisible(self._query_task.unified_login_flow2_enabled)
-        self.menu_action_ulf.setVisible(use_web)
-        self.menu_action_legacy.setVisible(not use_web)
+        self.menu_action_ulf.setVisible(can_use_web)
+        self.menu_action_legacy.setVisible(not can_use_web)
 
-        self.menu_action_ulf2.setEnabled(not self._use_local_browser)
-        self.menu_action_ulf.setEnabled(self._use_local_browser)
-        self.menu_action_legacy.setEnabled(self._use_local_browser)
+        self.menu_action_ulf2.setEnabled(
+            self.method_selected != auth_constants.METHOD_ULF2
```

Comment:
I got lost in the logic, but this makes me think that `method_selected` holds the current method, so should it be `==` instead? Otherwise, please disregard this comment.

Reply:
These `self.menu_action_` widgets are drop-down menu items. If the ULF2 method is already selected, we don't want to associated menu item to be enabled.

But we want the other ones to be so the user can actually change the method.